controllers/apply-next/form-router.js

```diff
@@ -36,6 +40,82 @@ function initFormRouter(formModel) {
+     */
+    router.get('/questions/:pdf?', (req, res) => {
+        const form = enhanceForm({
+            locale: req.i18n.getLocale(),
+            baseForm: formModel
+        });
+
+        const output = {
+            templates: {
+                html: path.resolve(__dirname, './views/questions-html.njk'),
+                pdf: path.resolve(__dirname, './views/questions-pdf.njk')
+            },
+            context: {
+                title: form.title,
+                form: form
+            }
+        };
+
+        if (req.params.pdf) {
+            const fileLocation = `documents/application-questions/${formModel.id}.pdf`;
+
+            const filePath = path.resolve(__dirname, '../../public/', fileLocation);
+
+            // First check to see if this file has already been rendered and saved in the app directory
+            fs.access(filePath, fs.F_OK, accessError => {
+                if (!accessError) {
+                    // The file exists so just redirect the user there
+                    return res.redirect(`/assets/${fileLocation}`);
+                }
+
+                // Otherwise it hasn't been rendered before, so we create it from scratch and save the file
+
+                // Repopulate existing global context so templates render properly
+                const context = { ...res.locals, ...req.app.locals, ...output.context };
+
+                // Render the HTML template to a string
+                nunjucks.render(output.templates.pdf, context, (renderErr, html) => {
+                    if (renderErr) {
+                        Raven.captureException(renderErr);
+                        res.status(400).json({ error: 'ERR-TEMPLATE-ERROR' });
+                    } else {
+                        // Turn HTML into a PDF
+                        pdf.create(html, {
+                            format: 'A4',
+                            base: config.get('domains.base'),
+                            border: '40px',
+                            zoomFactor: '0.7'
+                        }).toBuffer((pdfError, buffer) => {
+                            if (pdfError) {
+                                Raven.captureException(pdfError);
+                                return res.status(400).json({ error: 'ERR-PDF-BUFFER-ERROR' });
+                            }
+
+                            // Write the file locally so we can look it up next time instead of rendering
+                            fs.writeFile(filePath, buffer, writeError => {
+                                if (writeError) {
+                                    Raven.captureException(writeError);
+                                    return res.status(400).json({ error: 'ERR-PDF-WRITE-ERROR' });
+                                }
+                                // Give the user the file directly
+                                res.setHeader('Content-Disposition', `attachment; filename=${formModel.id}.pdf`);
```

Comment:
I think `res.download` does this for you https://expressjs.com/en/4x/api.html#res.download